development-process.md

```diff
@@ -0,0 +1,152 @@
+# ODL Development Process
+An approximate workflow guide to hacking on software at the MIT Office of Digital Learning.
+
+# Purpose
+Every team of developers has its own particular approach to writing, reviewing, and deploying code.
+This guide aims to capture MIT ODL's way of doing things. It will be (hopefully) useful as a
+step-by-step guide for new developers.
+
+# Tools
+
+- git
+- Zenhub Chrome extension (https://www.zenhub.io/)
+  - This extension adds the **Boards** and **Burndown** tabs to GitHub repos. We use it in the way
+  that you might use JIRA or Trello. The **Boards** tab is particularly important as it adds a
+  'swim lane'-type view for open issues.
+
+# Coding Workflow
+
+
+### Step 1) Hacking
+
+#### &#10146; Clone the given repository from GitHub
+
+Example: `git clone http://github.com/mitodl/micromasters.git`
+
+#### &#10146; Create a branch for the given issue
+
+Assuming you're working on issue **#123** in GitHub, you create a new branch from master like so:
+
+    git checkout -b NEW-BRANCH_NAME origin/master
+
+Eventually you'll want to push this new branch to origin to make it available to other developers.
+You can do this before you open a Pull Request (see below) or right after you create the branch:
+
+    git push -u origin NEW-BRANCH_NAME
+
+#### &#10146; Hack away!
+
+Make changes, test (automated and/or manual), commit, repeat. The repo README should have details
+for running automated tests.
+
+
+### Step 2) Code Review
+
+_When you've completed the given issue, the code is passing tests, and you feel like it's ready
+to ship, it's time to start code review. Make sure you read through the
+[Code Review guide](https://github.com/mitodl/handbook/blob/master/code_review.md) to get a sense
+about why we do code reviews and how to approach them, both as a reviewer and reviewee._
+
+#### &#10146; Open a Pull Request ("PR")
+
+This can be kicked off from several places in GitHub. The easiest one to remember is the "New
+Pull Request" button in your repo's "Pull Requests" tab
+([example](https://github.com/mitodl/micromasters/pulls)). Your **base** branch should be
+**master** and your **compare** branch should be your issue branch.
+
+When you first open the PR, the input should be pre-populated with our
+[PR template](https://github.com/mitodl/handbook/blob/master/pr-template.md). Fill in every section
+that is relevant to your PR. Under the **"relevant tickets"** section, be sure to indicate the
```

Comment:
People are not filling in every section.  We either do OR  -  a) specify which one are required - the first one should be AND b) delete the rest.